posts/d3-tutorial-points-axes.md

```diff
@@ -0,0 +1,139 @@
+
+           xAxisLength = width - 2 * margin;
+
+yAxisLength and xAxisLength determine the size of those axes, which is scaled to the height & width of the SVG item we're working within, respectively, while also taking into account the margin.
+
+    var xScale = d3
+             .scaleLinear()
+             .domain([0, lawfulness.length])
+             .range([0, xAxisLength]),
+           yScale = d3
+             .scaleLinear()
+             .domain([0, goodness.length])
+             .range([0, yAxisLength]),
+
+Here's where things start to get a bit fun! xScale and yScale create two linear scales from 0 to 4 (our max/min sizes), and set the Range to be from 0 to the length of that axis. Think of the range as a ruler, fixed in size, while "Domain" is the values on that ruler. For example, if I was to label each third inch on a ruler with it's own label (such as "A", "B", "C", "D"), then the domain would be 0-12 inches while the Range would be \["A", "B", "C", "D"\].
+
+    colorscale = d3
+             .scaleLinear()
+             .domain([min, max])
+             .interpolate(d3.interpolateHcl)
+             .range([d3.rgb("#0000ff"), d3.rgb("#8B0000")]); 
+
+Colorscale, meanwhile, takes a value from 0-4 and maps it to a color-HexCode from lightblue (0,0 or Lawful Good) to Dark Red (4,4 or Chaotic Evil). It does this by using D3's built-in interpolation libraries. The practical upshot of this is that the bottom-left of our graph will be very light blue, while the middle will be purple, and the top right will be dark red.
+
+## Rendering the Axes
+
+Now let's look at the renderXAxis method to learn how we rendered the Axes. Note that the Y Axis is incredibly similar, so we will only be going over the X.
+
+    var xAxis = d3
+             .axisBottom()
+             .ticks(lawfulness.length)
+             .tickFormat(t => {
+               return lawfulness[t];
+             })
+             .scale(xScale);
+
+First we create an Axis object oriented to the Bottom of the grid. We then give it a number of Ticks equal to the amount of items in our array, and format the label of that Tick to be the string value in our array. Then, we use the xScale we created earlier to map this 0-4 axis to the size of our xAxisLength.
+
+     svg.append("g")
+             .attr("class", "x-axis")
+             .attr("transform", function() {
+               return "translate(" + margin + "," + (height - margin) + ")";
+             })
+             .attr("opacity", 1)
+             .call(xAxis);
+
+Here, we are taking the SVG object and adding a new x-axis class object to it. The Transform method provided will position it, giving us a way to easily map other points to it later. We're setting the Opacity of this line to 100% (1) so that it appears bolder than the other gridlines, and passing the previously created "xAxis" object to the call method.
```

Comment:
```suggestion
Here, we are taking the SVG object and adding a new x-axis class object to it. The `transform` method provided will position it, giving us a way to easily map other points to it later. We're setting the `opacity` of this line to 100% (1) so that it appears bolder than the other gridlines, and passing the previously created `xAxis` object to the call method.
```